pkgs/development/python-modules/fmpy/default.nix

```diff
@@ -0,0 +1,153 @@
+    };
+
+    nativeBuildInputs = [ cmake ];
+  };
+
+  cvode = stdenv.mkDerivation {
+    pname = "cvode";
+    version = "5.3.0";
+
+    src = fetchFromGitHub {
+      owner = "LLNL";
+      repo = "sundials";
+      rev = "v5.3.0";
+      sha256 = "08q9svjdl6fmqrwyd9p12m98n0wd2lws38yh223wyp6590zi3i49";
+    };
+
+    nativeBuildInputs = [ cmake ];
+
+    cmakeFlags = [
+      "-DBUILD_ARKODE=OFF"
+      "-DBUILD_CVODES=OFF"
+      "-DBUILD_IDA=OFF"
+      "-DBUILD_IDAS=OFF"
+      "-DBUILD_KINSOL=OFF"
+      "-DEXAMPLES_ENABLE_C=OFF"
+    ];
+  };
+
+in
+
+buildPythonPackage {
+  pname = "FMPy";
+  version = "0.3.1";
+  disabled = pythonOlder "3.5";
+
+  src = fetchFromGitHub {
+    owner = "CATIA-Systems";
+    repo = "FMPy";
+    rev = "v0.3.1";
+    sha256 = "0ig08323id84nd9y272hyccky33my2nlhl44m41fxnkrg6xq7z3m";
+  };
+
+  nativeBuildInputs = [ cmake pyqt5 ];
+
+  propagatedBuildInputs = [
+    cmake
+    attrs
+    lark-parser
+    lxml
+    msgpack
+    numpy
+    scipy
+    pytz
+    dask
+    requests
+    matplotlib
+    pyqt5
+    pyqtgraph
+    notebook
+    plotly
+    rpclib
+    cvode
+  ];
+
+  dontUseCmakeConfigure = true;
+
+  postPatch = ''
+    substitute ${./libraries.py} ./fmpy/sundials/libraries.py \
+        --subst-var-by cvode ${cvode}
+  '';
+
+  # Don't run upstream build scripts as they are too specialized.
+  # And don't build sundials because it is already built.
+  preBuild =
+    let
+      platform = if stdenv.isDarwin then "darwin" else "linux";
+    in
+    ''
+      # Don't reproduce the full build_cvode.py, just two items:
+      # 1. The cswrapper (Model Exchange -> Co-Simulation wrapper).
+      cmake -DCVODE_INSTALL_DIR=${cvode} -S cswrapper -B cswrapper/build
+      cmake --build cswrapper/build --config Release
+
+      # 2. fmpy logger.
+      cmake -S fmpy/logging -B fmpy/logging/build
+      cmake --build fmpy/logging/build --config Release
+
+      # The reproduction of build_fmucontainer.py:
+      cmake -S fmpy/fmucontainer -B fmpy/fmucontainer/${platform}
+      cmake --build fmpy/fmucontainer/${platform} --config Release
+
+      # The reproduction of build_remoting.py
+      cmake -S remoting -B remoting/linux64 -D RPCLIB=${rpclib}
+      cmake --build remoting/linux64 --config Release
+    '';
+
+  # Some of tests are supposed to be failing in upstream, so we don't use
+  # them for package verification at the moment.
+  doCheck = false;
+
+  pythonImportsCheck = [
+    "fmpy"
+    "fmpy.cross_check"
+    "fmpy.cswrapper"
+    "fmpy.examples"
+    "fmpy.fmucontainer"
+    "fmpy.logging"
+    "fmpy.gui"
+    "fmpy.gui.generated"
+    "fmpy.ssp"
+    "fmpy.sundials"
+  ];
+
+  meta = with lib; {
+    description = "A free Python library to simulate Functional Mock-up Units (FMUs)";
+    homepage = "https://github.com/CATIA-Systems/FMPy";
+    license = with licenses; [ gpl3Only ];
+    maintainers = with maintainers; [ balodja ];
+    platforms = platforms.unix;
```

Comment:
```suggestion
```
that is the default for buildPythonPackage.

Reply:
OK, thank you!